tests/iterators.rs

```diff
@@ -555,11 +561,11 @@ fn test_fold() {
     a += 1;
     let mut iter = a.iter();
     iter.next();
-    assert_eq!(iter.fold(0, |acc, &x| acc + x), a.sum() - 1);
+    assert_eq!(iter.sum::<i32>(), a.sum() - 1);
```

Comment:
The `.sum()` implementation for `i32` *does* call `.fold()`, but since `test_fold` is intended to test `.fold()`, I think we should be explicit here (leave it as-is).